core/storage/ipfs/block.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_STORAGE_IPFS_BLOCK_HPP
+#define FILECOIN_STORAGE_IPFS_BLOCK_HPP
+
+#include "common/buffer.hpp"
+#include "common/cid.hpp"
+
+namespace fc::storage::ipfs {
+  /**
+   * @interface Piece of data, which is used with BlockService
+   */
+  struct Block {
+    using Content = common::Buffer; /**< Alias for content type */
+
+    /**
+     * @brief Get content identifier
+     * @return Block CID
+     */
+    virtual const CID &getCID() const = 0;
```

Comment:
Am I right that CID and Content can are properties of many different object types?
Maybe we should create additional interface for them and inherit Block from that interface?

Reply:
Tell me please, are you suggesting something like this? 

![CID_interface_v0](https://dl.dropboxusercontent.com/s/496x1n2t7z410pe/CID_interface_v0.svg)

Reply:
something like this